lib/internal/modules/esm/get_format.js

```diff
@@ -14,8 +14,9 @@ const { ERR_UNKNOWN_FILE_EXTENSION } = require('internal/errors').codes;
 const extensionFormatMap = {
   '__proto__': null,
   '.cjs': 'commonjs',
-  '.js': 'module',
-  '.mjs': 'module'
+  '.mjs': 'module',
+  '.json': 'unsupported',
+  '.wasm': 'unsupported'
```

Comment:
I wouldn't do these explicit opt-outs - the problem is that we can't predict what file extensions might need to be supported by the module system in future.